source/agora/node/BlockStorage.d

```diff
@@ -1180,9 +1180,32 @@ public class MemBlockStorage : IBlockStorage
     }
 }
 
+/// test memory storage
+unittest
+{
+    MemBlockStorage memory_storage = new MemBlockStorage();
+    testStorage(memory_storage);
+}
 
-///
+/// test disk storage
+// N.B. This is disabled by default as tests may become flaky if they rely on IO
+version (none)
 unittest
+{
+    import agora.utils.Test;
+
+    auto temp_dir = makeCleanTempDir(__MODULE__);
+    assert(temp_dir.exists);
+    scope(exit) mkdirRecurse(temp_dir);
+
+    auto diskStorage = new BlockStorage(temp_dir);
+    scope(exit) diskStorage.release();
+
+    testStorage(diskStorage);
+}
+
+version (unittest)
+void testStorage(IBlockStorage storage)
```

Comment:
Alright... I still don't think we need this PR, but if you feel strongly about it, just make this function:
```suggestion
private void testStorage (IBlockStorage storage)
```
and it will LGTM.

Reply:
Thanks, I do feel the test might be useful the next time someone has to update the block storage.

Reply:
You forgot the space  though

Reply:
:man_facepalming: added now.